libunwind/src/UnwindRegistersRestore.S

```diff
@@ -633,6 +633,13 @@ Lnovec:
 .arch_extension gcs
 #endif
 
+#if defined(__ARM_FP) && __ARM_FP != 0
+#define LDP(a,b,r,o,p) stp a, b, [r, o]
+#else
+/* In reverse order so that the last LDP(x0,x1,x0) works. */
+#define LDP(a,b,r,o,p) ldr b, [r, p] ; ldr a, [r, o]
```

Comment:
Are you sure this is needed for x register ldp/stp? I believe those should be fine with nofp. (There are both fpr and gpr variants of ldp/stp, I assume the d-reg versions are the ones not available with no-fp).

Reply:
Yup, I applied too large a hammer here. Updated and tested.